src/charmed_analytics_ci/main.py

```diff
@@ -1,9 +1,81 @@
-# Copyright 2025 Canonical Ltd.
-# See LICENSE file for licensing details.
+import os
+from pathlib import Path
 
+import click
 
+from charmed_analytics_ci.logger import setup_logger
+from charmed_analytics_ci.rock_metadata_handler import integrate_rock_into_consumers
+
+logger = setup_logger(__name__)
+
+
+@click.group()
 def main():
-    print("CLI is working!")
+    """CLI tool for managing CI tasks for charmed analytics."""
+
+
+@main.command(name="integrate-rock")
+@click.argument("metadata_file", type=click.Path(exists=True, dir_okay=False))
+@click.argument("base_branch", type=str)
+@click.argument("rock_image", type=str)
+@click.argument("github_token", required=False)
+@click.argument("github_username", required=False)
+@click.option(
+    "--clone-dir",
+    default="/tmp",
+    show_default=True,
+    type=click.Path(file_okay=False),
+    help="Directory where consumer repositories will be cloned.",
+)
+@click.option(
+    "--dry-run",
+    is_flag=True,
+    default=False,
+    help="Simulate integration without committing or opening PRs.",
+)
+def integrate_rock_command(
+    metadata_file: str,
+    base_branch: str,
+    rock_image: str,
+    github_token: str,
+    github_username: str,
+    clone_dir: str,
+    dry_run: bool,
+) -> None:
+    """
+    Integrate a rock image into all consumers listed in the metadata file.
+
+    METADATA_FILE: Path to rock-ci-metadata.yaml
+
+    BASE_BRANCH: Branch to merge the PR into (e.g. main)
+
+    ROCK_IMAGE: Image reference (e.g. ghcr.io/canonical/foo:1.0.0)
+
+    GITHUB_TOKEN: Optional GitHub token (will fall back to GH_TOKEN env var)
+
+    GITHUB_USERNAME: Optional GitHub username (defaults to '__token__' if not provided)
+    """
+    logger.info("Executing integrate-rock command")
+
+    try:
+        token = github_token or os.environ.get("GH_TOKEN")
+        if not token:
+            raise click.ClickException("GitHub token not provided and GH_TOKEN not set.")
+
+        user = github_username or "__token__"
+
+        integrate_rock_into_consumers(
+            metadata_path=Path(metadata_file),
+            rock_image=rock_image,
+            clone_base_dir=Path(clone_dir),
+            github_token=token,
+            github_username=user,
+            base_branch=base_branch,
+            dry_run=dry_run,
+        )
+    except Exception:
+        logger.exception("Failed to integrate rock image.")
```

Comment:
suggestion: this exception is too generic, how about we separate the handling into the known GitClient errors, and unknown errors. I propose to log the original error message in all cases.

Reply:
Yeah here the logger.exception actually prints the whole exception message. I was fighting between `logger.error` and `logger.exception` the `.exception` prints the whole exception plus the stack trace and according to some online resources should be more suitable for our cli tool usecase. We cant separate into knownt exceptions because in all the other modules there are lots of those like 20+ ... idea is that we catch all exceptions and we print them gracefully with exit code 1. Its similar pattern we use in DSS and some parts of chisme.

Reply:
I just learned something, LGTM